specs/language/resources.tex

```diff
@@ -0,0 +1,943 @@
+\end{HLSL}
+
+These each return the element at the given position \texttt{pos} of the type specified in the buffer definition.
+
+An additional \texttt{Load} method returns the indicated element just as the other,
+but also takes a second \texttt{status} parameter that returns information about the accessed resource.
+\begin{HLSL}
+ T Load(in int pos, out uint status);
+\end{HLSL}
+
+The \texttt{status} parameter returns an indication of whether the value retrieved accessed a mapped tile
+resource on tile-based platforms. This parameter is interpretted using the \texttt{CheckAccessFullyMapped}
+built-in function. If tiled resources are enabled, it returns true if the value was accessed from a mapped
+tile resource and false otherwise. If tiled resourced are not enabled, it will return true.
+
+Writable buffers have an additional subscript operator that allows assignment to an element of the buffer.
+It behaves as if it returns a reference to that element.
+\begin{HLSL}
+ T &operator[](uint pos);
+\end{HLSL}
+
+\Sec{Raw Buffer Types}{Resources.rawbufs}
+
+Raw buffers are either ByteAddressBuffers or StructuredBuffers.
+ByteAddressBuffers enable per-byte access to the raw data while
+StructuredBuffers have an associated structure type that determines how they are
+indexed. This type can be a scalar, vector, matrix, or user-defined struct.
+
+but partial writes aren't allowed.
+Though structured buffers can contain vectors,
+assignment to individual vector elements will result in an error.
+I HAVE BEEN UNABLE TO VERIFY THIS, THOUGH I WAS SURE IT WAS TRUE!
+
+\Sec{Read-only Byte Access Buffers}{Resources.robabufs}
+
+\begin{HLSL}
+ class ByteAddressBuffer {
+
+   ByteAddressBuffer();
+   ByteAddressBuffer operator=(ByteAddressBuffer buf);
+
+   void GetDimensions(out uint size);
+
+   // element access
+   uint Load(in uint byteOffset)
+   uint2 Load2(in uint byteOffset)
+   uint3 Load3(in uint byteOffset)
+   uint4 Load4(in uint byteOffset)
+   template<typename T>
+   T Load(in uint byteOffset);
+
+   uint Load(in uint byteOffset, out uint status)
+   uint2 Load2(in uint byteOffset, out uint status)
+   uint3 Load3(in uint byteOffset, out uint status)
+   uint4 Load4(in uint byteOffset, out uint status)
+   template<typename T>
+   T Load(in uint byteOffset, out uint status);
+};
+\end{HLSL}
+
+\Sub{Constructors}{Resources.robabufs.ctrs}
+
+Since their contents are viewed as raw bytes, ByteAddressBuffers are defined as a simple variable definition.
+\begin{HLSL}
+  ByteAddressBuffer buf;
+\end{HLSL}
+
+When defined at global scope, ByteAccessBuffers are bound to externally-defined backing stores
+using the explicit binding location if provided or the implicit binding if not.
+
+When defined at local scope, ByteAccessBuffers represent local references
+that can be associated with global ByteAccessBuffers when assigned,
+but must be resolvable to a unique global buffer.
+
+\begin{HLSL}
+  ByteAddressBuffer gbuf;
+  void doit(bool b) {
```

Comment:
> void doit(bool b)

Wondering why there's the boolean argument in this example?

Reply:
In error from when I had a more involved example. Removed.